enhancements/storage/csi-migration.md

```diff
@@ -133,12 +136,36 @@ spec:
 ```
 
 
-Once CSI migration of all storage plugins reaches GA in upstream, the associated feature gates will be enabled by default in OCP. As a result, we will remove the carry-patch we introduced in OCP 4.8. Also, users will not have to use the **TechPreviewNoUpgrade** *FeatureSet* anymore.
-
-### Risks and Mitigations
+#### Risks and Mitigations
 
 A carry-patch means that OCP will be the only Kubernetes distribution exercising this code path, which can lead us to bugs that were not seen anywhere. However, we are confident that the patch is small and self-contained enough to be used.
 
+### GA
+
+Once CSI migration of a given storage plugin reaches GA in upstream, the associated feature gates will be enabled by default in OCP. As a result, we will remove the carry-patch we introduced in OCP 4.8. Also, users will not have to use the **TechPreviewNoUpgrade** *FeatureSet* anymore.
+
+In addition to that, the CSI *StorageClass* (SC) will be set as default on new clusters, but we will not change the default SC on existing clusters. The main reason for that is that users might have set a different quota on the CSI driver SC. An extra benefit of this approach is that we will not override a default SC that was set by the user.
+
+In order to achieve that, these are the main changes introduced in OCP 4.11 with regards to SCs:
+
+* The *defaultstorageclasscontroller* from CSO will be completely removed and we will not create any new in-tree SCs.
+  * SCs that were previously created by this controller will not be owned by any operator and may very well be deleted by the user.
+* A new special-purpose controller will be added to the CSI Driver Operator.
+  * This controller will make sure that a CSI SC always exists.
+  * And it will set that SC as the default one when appropriate.
+  * Also, it will make sure there is only **one** default SC.
+
+In a more detaild way, this is how the new controller will operate and determine whether the CSI SC should be the default one:
+
+* On every sync, thie controller fetches all existing SCs.
+* If there are **no** pre-existing SCs in the cluster, then it creates a CSI SC **and** set it as the default.
```

Comment:
Can you clarify which "type" of SC it will look for?

Reply:
I would rephrase it to:

* If there are **no** pre-existing **default** SCs in the cluster, then it creates a CSI SC **and** set it as the default.
* Otherwise just create the CSI SC, but don't make it default.

Reply:
Done, but I think the **default**  above is unnecessary noise (if the are no pre-existing SCs, there can be no default SC anyway).

Reply:
But, there can be a case where there are existing SCs but they are not default and if we are going to create CSI storageclass then it should be default. So that is why we need to mention the default part.

Reply:
IMO we shouldn't make it default in that case. Otherwise, this would not only be a behavior change, but it could also break some use cases.

For example, a cluster admin set all SCs as **non** default to prevent users from using dynamic provisioning. I recall we even recommended this to customers, since we [intentionally don't overwrite the existing SC annotations](https://github.com/bertinatto/cluster-storage-operator/blob/f8eda8eeaba9ec2f275431e95555fa05a8c6fa81/pkg/operator/defaultstorageclass/controller.go#L173). What do you think? I'm really bad at taking notes during meetings, so I might have missed something (forgive me if that's the case).

Reply:
Yes - In that case we aren't creating any new SCs though. So that is what we talked about in the meeting, *only if* we are creating *new* SCs then we should mark it as default if there is no existing default exists in the cluster.

Reply:
Oh, I see... thanks for clarifying